From ccc2669f88fb06360262fc60cb3b25e919092294 Mon Sep 17 00:00:00 2001
From: Tatsuo ONISHI <onishi@latmos.ipsl.fr>
Date: Thu, 24 Nov 2022 09:06:08 +0100
Subject: [PATCH 44/71] (Update) Set lower limit for tsa24 in
 module_bioemi_megan2.F

TYPE : Update

Description:
       Lower limit for temperature (tsa) was already set,
       but it was not the case for temperature from the previous month (tsa24)
       Lower limit of tsa24 is set to 200K as was done for tsa.

List of modified files:
   * chem/module_bioemi_megan2.F
---
 chem/module_bioemi_megan2.F | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/chem/module_bioemi_megan2.F b/chem/module_bioemi_megan2.F
index 18229ce..c1dfafd 100644
--- a/chem/module_bioemi_megan2.F
+++ b/chem/module_bioemi_megan2.F
@@ -515,6 +515,11 @@ CONTAINS
           swd24 = mswdown (i,j,previous_month) ! [=] W/m2
 
           !...Perform checks on max and min bounds for temperature
+          IF (tsa24 .LT. 200.0) THEN
+             WRITE (mesg,'("temperature too low at i=",i3," ,j=",i3 )')i,j
+             CALL wrf_message(mesg)
+             if(tsa .ne. 0.0 .and. tsa24.eq.0.0) tsa24 = tsa
+          END IF
           IF (tsa .LT. 200.0) THEN
              WRITE (mesg,'("temperature too low at i=",i3," ,j=",i3 )')i,j
              CALL wrf_message(mesg)
-- 
2.31.1

